website/content/v1.4/advanced/etcd-maintenance.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "etcd Maintenance"
+description: "Operational instructions for etcd database."
+---
+
+`etcd` database backs Kubernetes control plane state, so `etcd` health is critical for Kubernetes availability.
+
+## Space Quota
+
+`etcd` default database space quota is set to 2 GiB by default.
+If the database size exceeds the quota, `etcd` will stop operations until the issue is resolved.
+
+This condition can be checked with `talosctl etcd alarm list` command:
+
+```bash
+$ talosctl -n <IP> etcd alarm list
+NODE         MEMBER             ALARM
+172.20.0.2   a49c021e76e707db   NOSPACE
+```
+
+If the Kubernetes database contains lots of resources, space quota can be increased to match the actual usage.
+The recommended maximum size is 8 GiB.
+
+To increase the space quota, edit the `etcd` section in the machine configuration:
+
+```yaml
+machine:
+  etcd:
+    extraArgs:
+      quota-backend-bytes: 4294967296 # 4 GiB
+```
+
+Once the node is rebooted with the new configuration, use `talosctl etcd alarm disarm` to clear the `NOSPACE` alarm.
+
+## Defragmentation
+
+`etcd` database can become fragmented over time if there are lots of writes and deletes.
+Kubernetes API server performs automatic compaction of the `etcd` database, which marks deleted space as free and ready to be reused.
+However, the space is not actually freed until the database is defragmented.
+
+If the database is heavily fragmented (in use/db size ratio is less than 0.5), defragmentation might increase the performance.
+If the database runs over the space quota (see above), but the actual in use database size is small, defragmentation is required to bring the on-disk database size below the limit.
+
+Current database size can be checked with `talosctl etcd status` command:
+
+```bash
+$ talosctl -n <CP1>,<CP2>,<CP3> etcd status
+NODE         MEMBER             DB SIZE   IN USE            LEADER             RAFT INDEX   RAFT TERM   RAFT APPLIED INDEX   LEARNER   ERRORS
```

Comment:
none are leaders, just the example output seems superfluous :sweat_smile:

Reply:
nope, the first one is the leader, its member ID matches known leader ID